Modules/_sre.c

```diff
@@ -248,22 +248,38 @@ data_stack_grow(SRE_STATE* state, Py_ssize_t size)
 /* -------------------------------------------------------------------- */
 /* factories and destructors */
 
-/* see sre.h for object declarations */
-static PyObject*pattern_new_match(PatternObject*, SRE_STATE*, Py_ssize_t);
-static PyObject *pattern_scanner(PatternObject *, PyObject *, Py_ssize_t, Py_ssize_t);
+/* module state */
+typedef struct {
+    PyTypeObject *Pattern_Type;
+    PyTypeObject *Match_Type;
+    PyTypeObject *Scanner_Type;
+} sre_module_state;
```

Comment:
I would prefer if you stick to common naming conventions:

* name the state after the module: ``_sremodulestate``
* name the state variable in functions ``state``, not ``module_state``

Reply:
The problem is that `_sre` already has a "SRE state", which has nothing to do with the module state. Perhaps we should rename the "SRE state" to `sre_state` (or `re_state`) in order to make the distinction more explicit.

Reply:
> * name the state after the module: `_sremodulestate`

Fixed in 44a4e47cdb

Reply:
> * name the state variable in functions `state`, not `module_state`

What do you think of the proposal above? (Renaming existing SRE state variables to `sre_state` or `re_state`, in order to make the distinction explicit.) It will explode the number of lines in this PR, but if you're fine with that, I'll get to it.